File: task/task_retrieve.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))


retrieved_task = client.task.retrieve("")

print(f"Task ID: {retrieved_task.id}")
print(f"Index ID: {retrieved_task.index_id}")
print(f"Video ID: {retrieved_task.video_id}")
print(f"Estimated time: {retrieved_task.estimated_time}")
print(f"Status: {retrieved_task.status}")
print("Metadata:")
for key, value in retrieved_task.metadata.items():
    print(f"  {key}: {value}")

if retrieved_task.hls:
    print("HLS:")
    print(f"  Video URL: {retrieved_task.hls.video_url}")
    print("  Thumbnail URLs:")
    for url in retrieved_task.hls.thumbnail_urls or []:
        print(f"    {url}")
    print(f"  Status: {retrieved_task.hls.status}")
    print(f"  Updated at: {retrieved_task.hls.updated_at}")

if retrieved_task.process:
    print("Process:")
    print(f"  Percentage: {retrieved_task.process.percentage}%")
    print(f"  Remaining Seconds: {retrieved_task.process.remain_seconds}")

print(f"Created at: {retrieved_task.created_at}")
print(f"Updated at: {retrieved_task.updated_at}")

File: task/task_retrieve.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY}); 
  const retrievedTask = await client.task.retrieve("")
  console.log(`Task ID=${retrievedTask.id}`);
  console.log(`Index ID: ${retrievedTask.indexId}`);
  console.log(`Video ID: ${retrievedTask.videoId}`);
  console.log(`Estimated time: ${retrievedTask.estimatedTime}`);
  console.log(`Status: ${retrievedTask.status}`);
  
  console.log("Metadata:");
  for (const [key, value] of Object.entries(retrievedTask.metadata)) {
    console.log(`  ${key}: ${value}`);
  }

  if (retrievedTask.hls) {
    console.log("HLS:");
    console.log(`  Video URL: ${retrievedTask.hls.videoUrl}`);
    console.log("  Thumbnail URLs:");
    for (const url of retrievedTask.hls.thumbnailUrls || []) {
      console.log(`    ${url}`);
    }
    console.log(`  Status: ${retrievedTask.hls.status}`);
    console.log(`  Updated at: ${retrievedTask.hls.updatedAt}`);
  }

  if (retrievedTask.process) {
    console.log("Process:");
    console.log(`  Percentage: ${retrievedTask.process.percentage}%`);
    console.log(`  Remaining Seconds: ${retrievedTask.process.remainSeconds}`);
  }

  console.log(`Created at: ${retrievedTask.createdAt}`);
  console.log(`Updated at: ${retrievedTask.updatedAt}`);
})();
